### PR TITLE
Seneca K107 checksum function

### LINKED ISSUE #400: Seneca K107 Checksum function
`senecak107.py` needs a checksum function written to check that data being read from the Seneca device hasn't been corrupted. The checksum formula hasn't been found in any of the device documentation so we may have to reverse-engineer the original C# code.

### PR DESCRIPTION
**Pull request by CWestICL**
Added a `calculate_crc()` function to the `SenekaK107` class that calculates the checksum from incoming data and compares it with the value at the end of the bytestring for validation. If the values don't match, an error is raised. The package 'crc' has also been added to the `poetry` environment.

Closes #400

### CODE REVIEW

**Comment by CWestICL** [comment]
Opened in draft as the data I have saved to test this function isn't passing the validation. I'm going into the lab this afternoon to test it with some fresh data and to double-check the original C# code the function is based on is functioning correctly.

**Review by dc2917 — CHANGES_REQUESTED** [review]
Some minor changes but otherwise looks good!

File: finesse/hardware/plugins/temperature/senecak107.py [hunk 1]
```diff
@@ -164,3 +188,9 @@ def get_temperatures(self) -> Sequence:
         self.request_read()
         data = self.read()
         return self.parse_data(data)  # type: ignore
+
```
[the remark this answers is not in the document]
Comment (CWestICL) [reply finesse/hardware/plugins/temperature/senecak107.py]: As it stands without the ignore statement, mypy throws out this error: `error: Incompatible return value type (got "ndarray[Any, Any]", expected "Sequence[Any]")`

[the remark this answers is not in the document]
Comment (dc2917) [reply finesse/hardware/plugins/temperature/senecak107.py]: Yeah, exactly. The return type annotation is wrong, it should be `numpy.ndarray`, not `Sequence`.

[the remark this answers is not in the document]
Comment (CWestICL) [reply finesse/hardware/plugins/temperature/senecak107.py]: Ah right gotcha

[the remark this answers is not in the document]
Comment (CWestICL) [reply finesse/hardware/plugins/temperature/senecak107.py]: Ok so I now remember why we used `Sequence` before, the `TemperatureMonitorBase` base class is using `Sequence` as the return type for the `get_temperatures()` function, so we get a differnet mypy error when using `numpy.ndarray`: `error: Return type "ndarray[Any, Any]" of "get_temperatures" incompatible with return type "Sequence[Any]" in supertype "TemperatureMonitorBase"`

[the remark this answers is not in the document]
Comment (dc2917) [reply finesse/hardware/plugins/temperature/senecak107.py]: Ah okay, yeah that makes sense. In that case, we should follow what the base class is doing and have `get_temperatures()` return a `Sequence`.
`return self.parse_data(data).tolist()` should do the trick.

**Review by dc2917 — APPROVED** [review]

**Pull request merged**